### ISSUE TITLE
Mesh surface roughness obtained using nerd data

### ISSUE THREAD

**Issue by Doraemon167** [NVlabs/nvdiffrec#18]
Thank you for your amazing work. I made a data set in nerd format. After training, the surface of mesh is not very good.  What parameters should I modify?
![20220415150739](https://user-images.githubusercontent.com/18532460/163537033-c499b522-2a87-467a-a83a-46cbb523aaf1.png)

My config is like this：
{
    "ref_mesh": "data/nerd/shoe",
    "random_textures": true,
    "iter": 20000,
    "save_interval": 100,
    "texture_res": [ 2048, 2048 ],
    "train_res": [960, 540],
    "batch": 4,
    "learning_rate": [0.03, 0.03],
    "kd_min" : [0.03, 0.03, 0.03],
    "kd_max" : [0.8, 0.8, 0.8],
    "ks_min" : [0, 0.08, 0],
    "ks_max" : [0, 1.0, 1.0],
    "dmtet_grid" : 128,
    "mesh_scale" : 3,
    "camera_space_light" : true,
    "background" : "white",
    "display" : [{"bsdf":"kd"}, {"bsdf":"ks"}, {"bsdf" : "normal"}],
    "out_dir": "shoe"
}

**Comment by jmunkberg** [comment]
Hello @Doraemon167 ,

The config looks reasonable. Depending on how your dataset is generated, you can perhaps remove `"camera_space_light" : true,` . If your lighting is constant in world space, you don't need that line. IIRC, we only used that setting in the `nerd_ehead.json` config. All other examples use the default value `"camera_space_light" : false,`
 
If your GPU allows for it (memory-wise), I would increase batch size to 8.  
For highest quality results, here is a checklist:
- Ensure that the lighting is constant in all training images
- Use nicely distributed views in the (hemi-)sphere, preferably 50+ views, and ensure that the object is mostly contained in the images.
- Double-check that the camera poses are accurate, and that the mapping from image to pose is correct.
- Use high quality foreground segmentation masks.

**Comment by Doraemon167** [comment]
Hi @jmunkberg Thanks for your reply, I did all your suggestions, but what does constant light mean? In my dataset, when the sun shines on the object surface, it will produce shadows on the other side. Does this have a negative impact on the texture or mesh? In addition, I want to try to experiment with higher resolution datasets. I don't know whether high-resolution images have a better effect on the results? Thank you again for your amazing work and reply.

**Comment by jmunkberg** [comment]
Thanks for your kind words!

With constant light, I mean that the positions of the light sources don't move while capturing the training data. We learn an environment map probe, and assume that this probe is constant, so a "morning to nighttime" capture setting is not ideal. Our code could likely be extended to capture dynamic light, but the current setup assumes constant lighting. 

Shadows will get baked into the albedo texture currently, as our renderer is a simple deferred renderer without explicit shadow support. This is usually ok (again, as long as the lighting is constant), but will reduce the quality in relighting scenarios. A more competent differentiable renderer, such as a differentiable ray tracer, will handle this, as discussed in the paper. 

High resolution images/training resolution will improve the result a bit, but will also increase the memory consumption linearly.  We usually run at 1k - 2k spatial resolution for high quality results, and use a batch size of 4-8. That is what fits into the memory of the 32-48GB GPUs we are using. With Torch DDP (`torchrun --nproc_per_node=4 train.py --config configs/bob.json`) you can increase the batch size further, if you have access to multiple GPUs.

**Comment by Doraemon167** [comment]
OK, thank you for your explanation. I got it. I will work according to your suggestions.

**Comment by 520jz** [comment]
> 

Hello friend, is your problem solved? I have a similar problem to you, if it is solved, can you tell me how to solve it?

**Comment by 520jz** [comment]
> OK, thank you for your explanation. I got it. I will work according to your suggestions.

Hello friend, is your problem solved? I have a similar problem to you, if it is solved, can you tell me how to solve it?
